presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestGroupInnerJoinsByConnectorRuleSet.java

```diff
@@ -0,0 +1,490 @@
+    }
+
+    @Test
+    public void testJoinPushDownHappenedWithFilters()
+    {
+        VariableReferenceExpression left = newBigintVariable("a1");
+        VariableReferenceExpression right = newBigintVariable("a2");
+        EquiJoinClause joinClause = new EquiJoinClause(left, right);
+
+        Set<JoinTableInfo> joinTableInfos = new HashSet<>();
+        Map<VariableReferenceExpression, ColumnHandle> assignments1 = ImmutableMap.of(newBigintVariable("a1"), new TestingColumnHandle("a1"), newBigintVariable("b1"),
+                new TestingColumnHandle("b1"));
+        List<VariableReferenceExpression> outputVariables1 = ImmutableList.of(newBigintVariable("a1"), newBigintVariable("b1"));
+        Map<VariableReferenceExpression, ColumnHandle> assignments2 = ImmutableMap.of(newBigintVariable("a2"), new TestingColumnHandle("a2"), newBigintVariable("b2"),
+                new TestingColumnHandle("b2"));
+        List<VariableReferenceExpression> outputVariables2 = ImmutableList.of(newBigintVariable("a2"), newBigintVariable("b2"));
+        JoinTableInfo joinTableInfo1 = new JoinTableInfo(new TestingMetadata.TestingTableHandle(new SchemaTableName(TEST_SCHEMA, TEST_TABLE)),
+                assignments1, outputVariables1);
+        JoinTableInfo joinTableInfo2 = new JoinTableInfo(new TestingMetadata.TestingTableHandle(new SchemaTableName(TEST_SCHEMA, TEST_TABLE)),
+                assignments2, outputVariables2);
+        joinTableInfos.add(joinTableInfo1);
+        joinTableInfos.add(joinTableInfo2);
+        JoinTableSet tableHandleSet = new JoinTableSet(joinTableInfos);
+        TableHandle tableHandle = new TableHandle(
+                new ConnectorId(CATALOG_SUPPORTING_JOIN_PUSHDOWN),
+                tableHandleSet,
+                TestingTransactionHandle.create(),
+                Optional.empty());
+
+        String expression = "a1 > b1";
+        TypeProvider typeProvider = TypeProvider.copyOf(ImmutableMap.of("a1", BIGINT, "b1", BIGINT));
+        TestingRowExpressionTranslator sqlToRowExpressionTranslator = new TestingRowExpressionTranslator(tester.getMetadata());
+        RowExpression rowExpression = sqlToRowExpressionTranslator.translateAndOptimize(expression(expression), typeProvider);
+
+        assertGroupInnerJoinsByConnectorRuleSet().on(p -> p.join(INNER,
+                        tableScan(CATALOG_SUPPORTING_JOIN_PUSHDOWN, "a1", "b1"),
+                        tableScan(CATALOG_SUPPORTING_JOIN_PUSHDOWN, "a2", "b2"),
+                        rowExpression,
+                        joinClause))
+                .matches(project(filter(
+                        "a1 = a2 and a1 > b1 and true",
+                        JoinTableScanMatcher.tableScan(CATALOG_SUPPORTING_JOIN_PUSHDOWN, tableHandle, "a1", "a2", "b1"))));
+    }
+
+    @Test
+    public void testPushDownWithTwoDifferentConnectors()
+    {
+        VariableReferenceExpression leftJoinKey1 = newBigintVariable("a1");
```

Comment:
It looks like a lot of these variables can be inlined. Please do that to make this code more readable